prow/git/v2/interactor.go

```diff
@@ -52,3 +62,264 @@ type Interactor interface {
 	// MergeCommitsExistBetween determines if merge commits exist between target and HEAD
 	MergeCommitsExistBetween(target, head string) (bool, error)
 }
+
+// CacheCloner knows how to clone repositories and update them for the central cache
```

Comment:
There is no central cache if a `from` has to be passed in or not?

Reply:
Good point. This interface is really two things -- the bit that knows how to create the cache and the other bit which knows how to use it. I split them out.

Reply:
Okay I am sorry but I do not really understand why these interfaces exist and why they are public. Unless I missed something, there is currently no way to get a `Cacher` or `Cloner` implementation, its only used within the private `bootstrapClients` which in turn is used by `ClientFromDir` and  and `ClientFor` which both return a `RepoClient`.

If those are only an internal implementation detail within the package, they should IMHO not be public and probably not even exist, unless we have tests that need them.